zephyr/src/supp_main.c

```diff
@@ -214,6 +226,12 @@ static int z_wpas_remove_interface(const char* ifname)
 		return -1;
 	}
 
+	if (z_wpas_get_iface_count() == 0) {
+		send_wifi_state_event(DEFAULT_IFACE_NAME, NET_EVENT_WPA_SUPP_CMD_NOT_READY);
+	}
+
+	send_wifi_state_event(ifname, NET_EVENT_WPA_SUPP_CMD_IFACE_REMOVED);
```

Comment:
This should be before sending not_ready, and I am also thinking if this should be send before we actually remove the interface? E.g., 

1. Removed interface
2. Application still send request on that iface
3. IFACE_REMOVED is sent

So, what will happen to #2, do we gracefully handle it? I guess even if we move #3 before #1 there could still be a race, but it narrows it.

Reply:
True. But I placed this at the end considering the case where we send the event but interface removal fails.

Reply:
Done